api/types.go

```diff
@@ -462,11 +462,76 @@ func (pT PinType) String() string {
 
 var pinOptionsMetaPrefix = "meta-"
 
+// PinMode is a PinOption that indicates how to pin something on IPFS,
+// recursively or direct.
+type PinMode int
+
+// PinMode values
+const (
+	PinModeRecursive PinMode = 0
+	PinModeDirect    PinMode = 1
+)
+
+// PinModeFromString converst a string to PinMode.
+func PinModeFromString(s string) PinMode {
+	switch s {
+	case "recursive", "":
+		return PinModeRecursive
+	case "direct":
+		return PinModeDirect
+	default:
+		logger.Warnf("unknown pin mode %s. Defaulting to recursive", s)
+		return PinModeRecursive
+	}
+}
+
+// String returns a human-readable value for PinMode.
+func (pm PinMode) String() string {
+	switch pm {
+	case PinModeRecursive:
+		return "recursive"
+	case PinModeDirect:
+		return "direct"
+	default:
+		return "recursive"
+	}
+}
+
+// MarshalJSON converts the PinMode into a readable string in JSON.
+func (pm PinMode) MarshalJSON() ([]byte, error) {
+	return json.Marshal(pm.String())
+}
+
+// UnmarshalJSON takes a JSON value and parses it into PinMode.
+func (pm *PinMode) UnmarshalJSON(b []byte) error {
+	var s string
+	err := json.Unmarshal(b, &s)
+	if err != nil {
+		return err
+	}
+	*pm = PinModeFromString(s)
+	return nil
+}
```

Comment:
For things like PinType, the API shows simply an integer which has not meaning whatsoever for the user. JSON serialization can use meaningful values though, so we do this here.